vendor/redisgreen.net/respio/resp.go

```diff
@@ -56,12 +56,20 @@ const (
 	INTEGER       = ':'
 	ARRAY         = '*'
 	ERROR         = '-'
-)
 
-var (
-	ErrInvalidSyntax = errors.New("resp: invalid syntax")
+	MAX_MESSAGE_LENGTH_IN_LOG = 200
 )
 
+func errInvalidSyntax(msg []byte) error {
+	var smsg string
+	if len(msg) > MAX_MESSAGE_LENGTH_IN_LOG {
+		smsg = string(msg[:MAX_MESSAGE_LENGTH_IN_LOG]) + "..."
+	} else {
+		smsg = string(msg)
+	}
+	return fmt.Errorf("resp: invalid syntax in %d:'%s'", len(msg), smsg)
```

Comment:
- okay, what's the intention behind printing "invalid syntax in {line_length_in_bytes}" ? Just nitpicking, but "in 123" suggests that 123 is an offset or something.
- I can see that this is always called with a `line`, so the error message will often end in `\r\n` (or even `\r...` if we're extremely lucky). Just checking - is it conventional to have newlines in errors? Won't this bork the log output? Also (since we're dealing with syntax errors) the line can potentially be binary junk -- maybe some golang equivalent of `repr(smsg)` would make sense here?

First golang review, I hope these aren't terribly stupid questions...

Reply:
Both very good points, thank you!  The first thing is a leftover from an experiment, the latter is just my not thinking this through well enough.
Fixed.
That's exactly why we do reviews, thanks :)